src/main/java/org/ethereum/beacon/discovery/DiscoverySystemBuilder.java

```diff
@@ -263,7 +263,7 @@ private void createDefaults() {
   LocalNodeRecordStore localNodeRecordStore;
   ExpirationSchedulerFactory expirationSchedulerFactory;
 
-  public DiscoverySystem build() {
+  public <T extends DiscoverySystem> T build(Class<T> desiredClass) {
```

Comment:
Extracting mutable/immutable interfaces looks like a good thing. 
Just not sure whether we need that complication to return the exact interface here. I would just return `MutableDiscoverySystem` here. 
@lucassaldanha what do you think?

Reply:
Yeah I realised I didn't submit my review comments earlier. My suggestion was to keep the build() method as is. So we don't break the API.

Potentially a separate buildMutable() method or something?